nodestream/pipeline/extractors/queues/extractor.py

```diff
@@ -0,0 +1,88 @@
+import json
+from abc import ABC, abstractmethod
+from logging import getLogger
+from typing import Any, Iterable
+
+from ....model import JsonLikeDocument
+from ....pluggable import Pluggable
+from ....subclass_registry import SubclassRegistry
+from ...flush import Flush
+from ..extractor import Extractor
+
+QUEUE_CONNECTOR_SUBCLASS_REGISTRY = SubclassRegistry()
+QUEUE_OBJECT_FORMAT_SUBCLASS_REGISTRY = SubclassRegistry()
+
+
+@QUEUE_CONNECTOR_SUBCLASS_REGISTRY.connect_baseclass
+class QueueConnector(Pluggable, ABC):
+    entrypoint_name = "queue_connectors"
+
+    @abstractmethod
+    async def poll(self) -> Iterable[Any]:
+        raise NotImplementedError
+
+
+@QUEUE_OBJECT_FORMAT_SUBCLASS_REGISTRY.connect_baseclass
+class QueueRecordFormat(Pluggable, ABC):
```

Comment:
I get `QueueConnector` and `QueueExtractor ` being a separate from stream extractor. By I am curious of `QueueRecordFormat` needs to be separate. The way I am thinking about it, a message is a message and on some level if it came from a queue based system or a stream based system is somewhat immaterial. I am curious what led you to separating this out and what your thoughts are here.

Reply:
I don't think it really needs to be I'll update